metadata_service/proxy/neo4j_proxy.py

```diff
@@ -1408,7 +1408,124 @@ def get_resources_using_table(self, *,
             results.append(DashboardSummary(**record))
         return {'dashboards': results}
 
+    @timer_with_counter
     def get_lineage(self, *,
-                    id: str,
-                    resource_type: ResourceType, direction: str, depth: int) -> Lineage:
-        pass
+                    id: str, resource_type: ResourceType, direction: str, depth: int = 1) -> Lineage:
+        """
+        Retrieves the lineage information for the specified resource type.
+
+        :param id: key of a table or a column
+        :param resource_type: Type of the entity for which lineage is being retrieved
+        :param direction: Whether to get the upstream/downstream or both directions
+        :param depth: depth or level of lineage information
+        :return: The Lineage object with upstream & downstream lineage items
+        """
+
+        def _get_badges(badges: list) -> List[Badge]:
```

Comment:
Just curious if there is somewhere else that this is being used or could be used? This seems like a reusable function across tables & lineage

Reply:
unrelated to other comment: how does `list` typecheck? Isn't `List` required?

Reply:
`List` is suitable for annotating return types. For arguments and other cases builtin `list` is fine. (or Iterable).

Reply:
> Just curious if there is somewhere else that this is being used or could be used? This seems like a reusable function across tables & lineage

@sewardgw good catch, refactored.